consensus/safety-rules/src/lib.rs

```diff
@@ -22,7 +22,7 @@ mod safety_rules_test;
 
 #[derive(Debug, Fail, Eq, PartialEq)]
 /// Different reasons for proposal rejection
-pub enum ProposalReject {
+pub enum Error {
```

Comment:
Let's not call our internal enum `Error`, cause this name is currently used by `failure`. As far as I understand you'd like to extend the enum because of the different types of rejects SafetyRules could return.

Let's go one step further and remove the `ProposalReject` enum at all. We're not really doing anything with the results today. Historically we had a lot of failure enums in the code, but at some point we realized we don't really need these structured errors if they're not changing anything for the caller.

Hence, I'd vote for removing `ProposalReject` at all and using `failure::Result` for the the functions (`bail!` if some of the safety rules don't match).

Reply:
This was my knee jerk reaction as well. Then I spent some time thinking about the synchronization problem. If the validator and the TCB are in sync, then there's likely not much the validator can do with an error other then say try again later. It could check another method say ... consensus_state() and try to rectify any disagreements but that's reactive rather than proactive. Perhaps we can save this nuance for a later commit?

Reply:
Oh, apparently @bmwill believes the right thing is to do exactly what you've done. Let it be.